vllm/model_executor/guided_logits_processors.py

```diff
@@ -170,3 +193,11 @@ def __init__(self, cfg: str, tokenizer: PreTrainedTokenizerBase):
         tokenizer = self.adapt_tokenizer(tokenizer)
         fsm = CFGFSM(cfg, tokenizer)
         self.fsm = fsm
+
+    def __deepcopy__(self, memo):
+        logits_processor = self.__class__(
+            copy.deepcopy(self.fsm.cfg_string, memo),
+            copy.deepcopy(self.fsm.tokenizer, memo),
```

Comment:
i'm confused why deepcopy of these objects are needed. both are immutable.

Reply:
Agreed